### ISSUE TITLE
kvserver/replicastats: NPE in (*ReplicaStats).RecordCount

### ISSUE THREAD

**Issue by mgartner** [cockroachdb/cockroach#94716]
Seen in [this `TestMultiRegionDataDriven/regional_by_row` CI failure](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/8187722?showRootCauses=true&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true).

```
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   |   GOROOT/src/runtime/panic.go:884
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes.func1
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_send.go:121
*   | runtime.gopanic
*   |   GOROOT/src/runtime/panic.go:890
*   | runtime.panicmem
*   |   GOROOT/src/runtime/panic.go:260
*   | runtime.sigpanic
*   |   GOROOT/src/runtime/signal_unix.go:835
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicastats.(*ReplicaStats).RecordCount
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicastats/replica_stats.go:264
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver/load.(*ReplicaLoad).record
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/load/replica_load.go:122
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver/load.(*ReplicaLoad).RecordBatchRequests
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/load/record_replica_load.go:18
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).recordBatchRequestLoad
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_send.go:1026
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).SendWithWriteBytes
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_send.go:141
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_send.go:205
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/stores.go:203
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal
*   |   github.com/cockroachdb/cockroach/pkg/server/node.go:1121
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch
*   |   github.com/cockroachdb/cockroach/pkg/server/node.go:1173
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:741
*   | github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ServerInterceptor.func1
*   |   github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:95
*   | github.com/cockroachdb/cockroach/pkg/rpc.bindUnaryServerInterceptorToHandler.func1
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:813
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:260
*   | github.com/cockroachdb/cockroach/pkg/rpc.bindUnaryServerInterceptorToHandler.func1
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:813
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:73
*   | github.com/cockroachdb/cockroach/pkg/rpc.bindUnaryServerInterceptorToHandler.func1
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:813
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:229
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   |   github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:322
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:227
*   | github.com/cockroachdb/cockroach/pkg/rpc.bindUnaryServerInterceptorToHandler.func1
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:813
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:751
*   | github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ClientInterceptor.func2
*   |   github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:225
*   | github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:897
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:782
*   | github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch
*   |   github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:905
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:209
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext
*   |   github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:188
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```

Jira issue: CRDB-23098

**Comment by mgartner** [comment]
Might this be related to #94696?

**Comment by kvoli** [comment]
resolved on #94709
